x-pack/plugins/security_solution/public/common/components/header_actions/actions.tsx

```diff
@@ -234,11 +243,14 @@ const ActionsComponent: React.FC<ActionProps> = ({
     [documentBasedNotes, timelineNoteIds, securitySolutionNotesEnabled]
   );
 
-  const noteIds = useMemo(() => {
-    return securitySolutionNotesEnabled
-      ? documentBasedNotes.map((note) => note.noteId)
-      : timelineNoteIds;
-  }, [documentBasedNotes, timelineNoteIds, securitySolutionNotesEnabled]);
+  /* note ids specific to the current timeline, it is used to enable/disable pinning */
```

Comment:
@christineweng ,

We do need to use this to enable/disable pinning, since we already have `disablePinning` param which should be used by parent component if we want to effect the visibility of `pin` action.